### PR TITLE
Cleanup ability helpers

### PR DESCRIPTION
**Pull request by mbj**
To get the flay threshold down to something meaningful.

### CODE REVIEW

File: core/spec/models/spree/ability_spec.rb [hunk 1]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
 end
 
 describe Spree::Ability, :type => :model do
+  shared_examples_for 'access granted' do
+    it 'should allow read' do
+      expect(ability).to be_able_to(:read, resource, token) if token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: I wonder if we could replace this conditional using:

```ruby
expect(ability).to be_able_to(:read, resource, *token)
```

File: core/spec/models/spree/ability_spec.rb [hunk 2]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
...
+      expect(ability).to be_able_to(:read, resource) unless token
+    end
+
+    it 'should allow create' do
+      expect(ability).to be_able_to(:create, resource, token) if token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: Instance of https://github.com/MountainRoseHerbs/spree/pull/399/files#r69246772

File: core/spec/models/spree/ability_spec.rb [hunk 3]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
...
+      expect(ability).to be_able_to(:create, resource) unless token
+    end
+
+    it 'should allow update' do
+      expect(ability).to be_able_to(:update, resource, token) if token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: Instance of https://github.com/MountainRoseHerbs/spree/pull/399/files#r69246772

**Comment by dkubb** [comment]
@mbj I left a couple of notes on this PR.

**Comment by mbj** [comment]
@dkubb Ready for the next pass

File: core/spec/models/spree/ability_spec.rb [hunk 4]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
 end
 
 describe Spree::Ability, :type => :model do
+  shared_examples_for 'access granted' do
+    it 'should allow read' do
+      expect(ability).to be_able_to(:read, resource, *token)
+      expect(ability).to be_able_to(:read, resource) unless token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: I think these `unless token` statements are unnecessary.

If the line above has a `nil` token, then `*nil` will cause nothing to be passed into the method, making it equivalent to this line.

File: core/spec/models/spree/ability_spec.rb [hunk 5]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
...
+    end
+
+    it 'should allow create' do
+      expect(ability).to be_able_to(:create, resource, *token)
+      expect(ability).to be_able_to(:create, resource) unless token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: Instance of https://github.com/MountainRoseHerbs/spree/pull/399/files#r69487025

File: core/spec/models/spree/ability_spec.rb [hunk 6]
```diff
@@ -17,6 +16,81 @@ def initialize(user)
...
+    end
+
+    it 'should allow update' do
+      expect(ability).to be_able_to(:update, resource, *token)
+      expect(ability).to be_able_to(:update, resource) unless token
```
Comment (dkubb) [inline core/spec/models/spree/ability_spec.rb]: Instance of https://github.com/MountainRoseHerbs/spree/pull/399/files#r69487025

**Comment by dkubb** [comment]
@mbj I left a few notes on this PR.

**Pull request merged**